File: run_shadow.py
#!/usr/bin/env -S python3 -B

from time import time
from common.tk_drawer import TkDrawer
from shadow.polyedr import Polyedr


tk = TkDrawer()
try:
    for name in ["ccc", "cube", "box", "king", "cow"]:
        print("=============================================================")
        print(f"Начало работы с полиэдром '{name}'")
        start_time = time()
        p = Polyedr(f"data/{name}.geom")
        # Печать характеристики: сумма длин рёбер с «хорошими» концами
        good_sum = p.good_edges_length_sum()
        print(f"Сумма длин рёбер с «хорошими» концами: {good_sum}")
        p.draw(tk)
        delta_time = time() - start_time
        print(f"Изображение полиэдра '{name}' заняло {delta_time} сек.")
        input("Hit 'Return' to continue -> ")
except(EOFError, KeyboardInterrupt):
    print("\nStop")
    tk.close()

File: c_src/run_shadow.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include "common/tk_drawer.h"
#include "shadow/polyedr.h"

int main(void) {
    const char *names[] = {"ccc", "cube", "box", "king", "cow"};
    int num_names = 5;
    
    TkDrawer *tk = tk_create();
    if (!tk) {
        fprintf(stderr, "Не удалось создать графический интерфейс\n");
        return 1;
    }
    
    for (int i = 0; i < num_names; i++) {
        printf("=============================================================\n");
        printf("Начало работы с полиэдром '%s'\n", names[i]);
        
        char filename[256];
        snprintf(filename, sizeof(filename), "../data/%s.geom", names[i]);
        
        clock_t start_time = clock();
        
        Polyedr *p = polyedr_create(filename);
        if (!p) {
            fprintf(stderr, "Не удалось загрузить полиэдр из файла '%s'\n", filename);
            continue;
        }
        
        polyedr_draw(p, tk);
        
        clock_t end_time = clock();
        double delta_time = (double)(end_time - start_time) / CLOCKS_PER_SEC;
        
        printf("Изображение полиэдра '%s' заняло %f сек.\n", names[i], delta_time);
        
        /* Ожидание нажатия клавиши */
        printf("Hit 'Return' to continue -> ");
        fflush(stdout);
        getchar();
        
        polyedr_free(p);
    }
    
    tk_close(tk);
    printf("\nStop\n");
    
    return 0;
}
